core/whistle/src/wh_json.erl

```diff
@@ -117,53 +117,38 @@ decode(JSON, <<"application/json">>) ->
     try jiffy:decode(JSON) of
         JObj -> JObj
     catch
-        'throw':{'error',{_Loc, 'invalid_string'}} ->
+        'throw':{'error',{_Loc, 'invalid_string'}}=E ->
             lager:debug("invalid string(near char ~p) in input, checking for unicode", [_Loc]),
-            try_converting(JSON);
-        'throw':{'invalid_json',{{'error',{1, _Err}}, _Text}} ->
-            lager:debug("~s: ~s", [_Err, _Text]),
-            new();
-        'throw':{'invalid_json',{{'error',{Loc, _Err}}, Text}} ->
+            try_converting(JSON, E);
+        'throw':{'invalid_json',{{'error',{1, Err}}, _Text}} ->
+            lager:debug("~s: ~s", [Err, _Text]),
+            throw({'error', Err});
+        'throw':{'invalid_json',{{'error',{Loc, Err}}, Text}} ->
             Size = erlang:byte_size(Text),
             Part = binary:part(Text, Loc, Size-Loc),
-            lager:debug("~s near char # ~b of ~b ('~s'): ~s", [_Err, Loc, Size, Part, Text]),
-            log_big_binary(iolist_to_binary(JSON)),
-            try_converting(JSON);
-        'throw':_E ->
-            lager:debug("thrown decoder error: ~p", [_E]),
-            new();
-        _Error:_Reason ->
-            lager:debug("decoder exited with ~p:~p", [_Error, _Reason]),
-            log_big_binary(iolist_to_binary(JSON)),
```

Comment:
The intent of https://github.com/2600hz/kazoo/pull/1668 was to log the impeding JSON that made some kazoo app crash without explanation. By logging here we make sure that next time it happens, we have logs to inspect